### DISCUSSION TITLE
DBeaver Export Command

### DISCUSSION THREAD

**Discussion by id10t-errors** [dbeaver/dbeaver#17172] [Q&A]
Trying to use the export function in DBeaver. I'm not understanding how to implement the settings where I do not want the default. All I want to do is pick the filepath and file name. Can someone please help?

here's what I have:
@export {"type": <csv>, "producer": {}, "consumer": {<outputFolder='filepath' outputFilePattern = 'data'>}, "processor": {}}

https://dbeaver.com/docs/wiki/Export-Command/

**Comment by LonwoLonwo** [comment]
@ShadelessFox

**Comment by ShadelessFox** [comment]
Hello @error-id10t,

The command should look like this:

```
@export {"type": "csv", "producer": {}, "consumer": {"outputFolder": "filepath", "outputFilePattern": "data"}, "processor": {}}
```

Does it yield the desired result?

**Reply by id10t-errors** [reply]
Unfortunately not. It returns: Invalid syntax. Use '@export {"type": <type>, "producer": {...}}, "consumer": {...}}, "processor": {...}}'

I tried putting the filepath right to my C drive to keep it as simple as possible. I should have mentioned I'm on v22.1.0, since DBeaver's documentation seems to note syntax varies by version. Is the function intended to replicate the export function, so it will just export the current results and I do not need to specify what is being output?

Thanks!

**Reply by ShadelessFox** [reply]
Please provide your command arguments

**Reply by ladeche** [reply]
Hello @ShadelessFox,

Here's an example on Windows that doesn't fire any error but doesn't do anything : 

`@export { "type": "csv", "producer": {}, "consumer": { "outputFolder":"C:\Tmp", "outputFilePattern":"output"}, "processor": {"extension":"csv", "delimiter":"|"}  }
`
I get this as a result ("Statistic tab") :
`
Updated Rows	-1
Query	@export { "type": "csv", "producer": {}, "consumer": { "outputFolder":"C:\Tmp", "outputFilePattern":"output"}, "processor": {"extension":"csv", "delimiter":"|"}  }
Finish time	Fri Sep 16 16:54:54 CEST 2022
`

But it doesn't produce any file. (no C:\Tmp\output.csv file)

The thing is I don't understand how this command knows what (ie which query) to export ? Latest query ?

Thanks.

**Reply by ShadelessFox** [reply]
Hello @ladeche,

The command itself doesn't do anything, it behaves like a marker for regular queries. You will need to run any other SQL query afterwards in order to trigger the command.

Additionally, in our example, you need to escape the backslash symbol in the `outputFolder` parameter like so: `"C:\\Tmp"`.

Please let me know if you have any questions left!

**Reply by ladeche** [reply]
Thank you @ShadelessFox for your response.

Now, I'm not sure I get it. Here's my code in the SQL editor :
`
@export { "type": "csv", "producer": {}, "consumer": { "outputFolder":"C:\\Tmp", "outputFilePattern":"output"}, "processor": {"extension":"csv", "delimiter":"|"} 
SELECT * FROM mytable;
`
If I understand correctly what you said, I should get the output of the query in the C:\Tmp\output.csv file. 
But when I execute this as a script, I get 2 tabs as a result. One with the content of mytable, the other with statistics as in my previous message. But, still no C:\Tmp\output.csv file.

What am I missing ? By the way I'm working with release 21.1.3.

**Reply by ShadelessFox** [reply]
Oh, I see. This feature was implemented in 22.0.4, so you'll have to upgrade to a newer version.

**Comment by ladeche** [comment]
Hello @ShadelessFox 

Thank you for your reply. I installed DBeaver 22.2 and now it works, although not exactly as I expected. So, when the query in encountered the export wizard opens with a predetermined config. 

My goal was that the result of query would be exported directly into the file without any interaction. Also I was willing to chain exports in a single script such as : 
```SQL
@export {...file1}
SELECT ...;

@export{...file2}
SELECT ...;
```
Do you know if this is possible ?

If not I will stick with a task starting a "PostgreSQL script" which uses psql.

**Reply by ShadelessFox** [reply]
It's not possible using the `@export` command. Its purpose is just to open the wizard.

You can click on a query, then use <kbd>Execute</kbd> &rArr; <kbd>Export from Query</kbd> from the context menu to create an export task.

**Comment by voronczoff** [comment]
Is @export command supported in Community version of DBeaver?

**Reply by LonwoLonwo** [reply]
Yes.

**Reply by vladunev-tomtom** [reply]
Have you actually tried it?

**Reply by ShadelessFox** [reply]
https://github.com/dbeaver/dbeaver/wiki/Export-Command

If it doesn't work for you for some reason, please create a new issue with detailed steps required to reproduce it.

**Reply by vladunev-tomtom** [reply]
Ah, maybe "prefills values" insted of "allows you to open" would be easier to understand? 
I was expecting that export dialog will open and save me all the repetetive clicks in the Export Dialog.

**Reply by ShadelessFox** [reply]
Well, it says both:
> allows you to open the data transfer wizard with prefilled settings

But yes, the whole purpose of it is to prerefill values for you.
